.github/workflows/protocol-devchain-anvil.yml

```diff
@@ -124,17 +123,35 @@ jobs:
       - name: 'Install packages'
         shell: bash
         run: yarn
-
-      - name: Generate migrations and run devchain
+          
+      # Install `lsof` dependency, because it's not readily available on CI, but is required by 
+      # `create_and_migrate_anvil_l2_devchain.sh`, because it uses `stop_anvil.sh` to kill 
+      # existing anvil servers.
+      - name: Generate L1 and L2 migrations
         if: success() || failure()
-        run: ./scripts/foundry/create_and_migrate_anvil_devchain.sh
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y lsof
+          
+          ./scripts/foundry/create_and_migrate_anvil_l2_devchain.sh
 
-      - name: Run integration tests against local anvil devchain
+      - name: Run integration tests against L1 devchain
         if: success() || failure()
         run: |
+          source ./scripts/foundry/constants.sh
+
+          anvil \
+          --port 8547 \
+          --state $TMP_FOLDER/$L1_DEVCHAIN_FILE_NAME \
+          --state-interval $STATE_INTERVAL \
+          --gas-limit $GAS_LIMIT \
+          --code-size-limit $CODE_SIZE_LIMIT \
+          --balance $BALANCE \
+          --steps-tracing &
+
           forge test -vvv \
           --match-path "test-sol/integration/*" \
-          --fork-url http://127.0.0.1:${{ env.ANVIL_PORT }}
+          --fork-url http://127.0.0.1:8547
```

Comment:
There is a bug with this workflow on CI:

```
2024-07-08T09:46:34.824960Z ERROR foundry_evm_core::fork::init: It looks like you're trying to fork from an older block with a non-archive node which is not supported. Please try to change your RPC url to an archive node if the issue persists.
Error: 
Could not instantiate forked environment with fork url: http://127.0.0.1:8547/
Context:
- Failed to get block for block number: 256
latest block number: 256
```

Source: [CI](https://github.com/celo-org/celo-monorepo/actions/runs/9837180487/job/27154422476?pr=11107#step:20:599)

Reply:
I'm still getting this bug as of this [CI run](https://github.com/celo-org/celo-monorepo/actions/runs/9838934169/job/27159898549?pr=11107) in this commit https://github.com/celo-org/celo-monorepo/commit/cd6ca0544b8d70a4c18f7393d3e3965829485260.

Reply:
As of a906048, I implemented a workaround to that passes CI. 

Basically, I'm generating and running an L1 devchain from scatch, and use that to run the integration test.

This takes longer than simply starting an L1 devchain from JSON state, but prevents the Anvil bug when loading state:
- foundry-rs/foundry#7502

Reply:
I'll mark this as resolved but will leave a note for future reference. So this can be refactored when the bug is merged and we can bug the version.